### DISCUSSION TITLE
how to reduce the size of a repository  ??

### DISCUSSION THREAD

**Discussion by HDInfautre** [borgbase/vorta#1946] [FAQ]
Hello
**how to reduce the size of a repository when the storage is = 180gb and I would like it to be between 60gb and 100gb ?**

For example, I limited it to 181GB. I restarted the backup but I got an error message
"This is to let you know that you are currently using 100% of your quota for repo ComInfxxxx. If you plan to add more files, you should increase the quota or future backups may fail. "
I would like it to delete old backups

**Comment by HDInfautre** [comment]
I set a slightly higher limit of 189GB to the currently stored value of 181GB.
I deleted old archives. I still get the error message from the backup about exceeding the maximum value.
I don't want to delete manually every time, but have borgbase do this automatically whenever it cannot exceed the max storage value of the repository.

**Comment by sayem314** [comment]
Check https://vorta.borgbase.com/usage/prune/

**Reply by goebbe** [reply]
@HDInfautre  I also believe automatic **pruning** is the way to go. 
However, to my knowledge Vorta/ borg has no automatic mechanism that prunes old archives depending on the size of the repository. So you have to check/ guess how many copies of your backups you can keep on your backup server and adapt the pruning rules accordingly. 

If you delete archives, don't forget to apply "**compact**" to free disk space.  
In Vorta you can find **compact** in the Archives tab, at the bottom of the list of archives. 

From the borg docs (Vorta uses borg): 
**compact** frees repository space by compacting segments.
Use this regularly to avoid running out of space - you do not need to use this after each command though. It is especially useful after deleting archives, because only compaction will really free repository space.
[link to borg docs](https://borgbackup.readthedocs.io/en/stable/usage/compact.html)

If you want to run compact automatically in Vorta you probably could (ab)use Vorta>Schedule> Shell commands> pre/ post backup to run the `borg compact` on a regular schedule for you repository.

**Reply by HDInfautre** [reply]
Ok thanks
https://github.com/borgbase/vorta/issues/1951

**Comment by goebbe** [comment]
@HDInfautre  
I assume you use automatic (scheduled) backups? 
To set up **pruning** you can:  
- In Vorta > Schedule:  enable autopruning  (Prune after each backup)
- In Vorta > Archives > Prune options: Set how many backups you want to keep e.g. 2 daily and 2 monthly
  This will keep e.g. the last two daily backups and keep the last two monthly backups. All other archives will be pruned (i.e. deleted) automatically.

This should work pretty well for most use cases, if you have only a modest amount of file changes between the backups. 
It is probably sufficient to **compact** your archives manually twice a year.
